src/components/Primary-Button/Button.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import { DEFAULT_PROPS, SIZE_CONFIGS } from './Button.constants';
+import { StyledButton, ButtonContent, LoadingSpinner } from './Button.styles';
+
+import type { ButtonProps } from './Button.types';
+
+export const PrimaryButton: React.FC<ButtonProps> = ({
```

Comment:
The component is typed using React.FC<ButtonProps>. The project prefers explicit prop typing on the function signature (avoid React.FC) to avoid React.FC implicit children and other caveats.

Suggestion:
- Use: export const PrimaryButton = ({ ... }: ButtonProps) => { ... } instead of React.FC<ButtonProps>.
- This keeps props explicit and avoids potential surprises with defaultProps/children typing.